kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -477,26 +521,32 @@ private static void validatePhysicalNameConsistency(
     }
   }
 
-  /* Validate if a given schema evolution is safe for a given column mapping mode*/
-  private static void validateSchemaEvolution(
+  /*
+   * Validate if a given schema evolution is safe for a given column mapping mode
+   *
+   * <p>Returns an updated schema if metadata (i.e. TypeChanges needs to be copied
+   * over from currentSchema).
+   */
+  private static Optional<StructType> validateSchemaEvolution(
```

Comment:
nit: not it is doing more than validation, may be rename to validateSchemaEvolutionAndGetUpdatedSchema

Reply:
I am actually not sure why we need to update the schema? are we expecting the client to just do type changes and we need to generate the appropriate schema struct filed changes?